snowflake/data_access.go

```diff
@@ -672,46 +698,46 @@ func (s *AccessSyncer) generateAccessControls(ctx context.Context, apMap map[str
 				}
 
 				if isTableType(what.DataObject.Type) {
-					grants, err := s.createGrantsForTableOrView(what.DataObject.Type, permissions, what.DataObject.FullName, metaData)
-					if err != nil {
-						return err
+					grants, err2 := s.createGrantsForTableOrView(what.DataObject.Type, permissions, what.DataObject.FullName, metaData)
+					if err2 != nil {
+						return s.handleAccessProviderFeedback(feedbackHandler, &fi, err2)
```

Comment:
Instead of doing this everywhere we could move the statements inside the for loop to a dedicated method. Within that method we can return an error and handle the execute the handleAccessProviderFeedback outside that method (or in a defer). In that case there is no reason to execute this on every return.